solar_system3.rb

```diff
@@ -0,0 +1,303 @@
+def add_planet(solar_system)
+  # get info from user
+  new_planet = ask_for_info
+
+  add_new_planet(solar_system, new_planet)
+
+  # go back to main menu
+  return return_to_main_menu
+
+end
+
+# prompts user for info to create a planet
+def ask_for_info
+  puts "Please provide the following information"
+
+  print "\nPlanet name? "
+  name = gets.chomp.capitalize
+
+  print "Diameter (in km)? "
+  diameter = gets.chomp.to_f
+
+  print "Mass (in kg)? "
+  mass = gets.chomp.to_f
+
+  print "Number of moons? "
+  num_moons = gets.chomp.to_i
+
+  print "Distance from the sun (in km)? "
+  distance_from_the_sun = gets.chomp.to_f
+
+  print "Year length (in days)? "
+  year_length = gets.chomp.to_f
+
+  new_planet = Planet.new(name, diameter, mass, num_moons, distance_from_the_sun, year_length)
+
+  return new_planet
+
+end
+
+# creates planet from user input
+def create_planet(solar_system)
+  puts "\nMAKE-A-PLANET"
+
+  new_planet = ask_for_info
+
+  puts "\nPlanet created!"
+  puts "\n" + new_planet.return_planet_info
+
+  if add_to_solar_system?
+    add_new_planet(solar_system, new_planet)
+  end
+
+  return return_to_main_menu
+
+end
+
+# method that prompts user to learn about one of the planets
+def learn(solar_system)
+  puts "What planet would you like to learn about?"
+  puts "\n"
+  puts solar_system.return_planet_list
+  puts "Type planet name or 'exit' to go back to the main menu"
+
+  choice = gets.chomp.capitalize
+
+  while choice != "Exit" && choice != ""
+    selected_planet = select_planet(solar_system, choice)
+    if selected_planet == nil
+      puts "#{choice.capitalize} is not in the solar system. Please choose another planet or type 'exit' to go back to the main menu"
+      choice = gets.chomp.capitalize
+    else
+      puts solar_system.get_info(selected_planet)
+
+      puts "\nWhat planet would you like to learn about now?"
+      choice = gets.chomp.capitalize
+    end
+  end
+
+  return return_to_main_menu
+  # main_menu
+  # choice = gets.chomp.capitalize
+  # return choice
+end
+
+mercury = Planet.new("Mercury", 4879.4, 3.3e23, 0, 57_909_175, 87.97)
+venus = Planet.new("Venus", 12_104, 4.87e24, 0, 108_208_930, 224.7)
+earth = Planet.new("Earth", 12_756, 5.98e24, 1, 149_597_890, 365.26)
+mars = Planet.new("Mars", 6787, 6.42e23, 2, 227_936_640, 686.98)
+jupiter = Planet.new("Jupiter", 142_800, 1.90e27, 53, 778_412_020, 4331.98)
+saturn = Planet.new("Saturn", 120_660, 5.69e26, 53, 1_426_725_400, 10_760.56)
+uranus = Planet.new("Uranus", 51_118, 8.68e25, 27, 2_870_972_200, 30_685)
+neptune = Planet.new("Neptune", 49_528, 1.02e26, 13, 4_498_252_900, 60_190)
+
+planet_list = [mercury, venus, earth, mars, jupiter, saturn, uranus, neptune]
+
+ss = SolarSystem.new(planet_list, 4.6e9)
+
+def view_solar_system(solar_system)
+  puts intro(solar_system)
+
+  main_menu
+  choice = gets.chomp.downcase
+
+  while choice != "exit" && choice != ""
+
+    case choice.to_i
+    when 1
+      choice = learn(solar_system)
```

Comment:
Small bug: If you first make a choice to "Learn" with "1", and then you try to exit with "exit", it says that "exit" is an invalid selection.